Lab 1.2.1/Lab 1.2.1/LibraryUser.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Lab_1._2._1
+{
+    public interface ILibraryUser
+    {
+        void AddBook(string bookName);
+        void RemoveBook(string bookName);
+        string GetBookInfo(int index);
+        int GetBooksCount();
+    }
+
+    public class LibraryUser : ILibraryUser
+    {
+        private static int nextId;
+        public static int GetNextId()
+        {
+            return nextId++;
+        }
+
+        public string Phone { get; set; }
+        public int BookCount { get => this.bookList.Count; }
+
+        public readonly string FirstName;
+        public readonly string LastName;
+        public readonly int Id;
+        public readonly int BookLimit;
+        public readonly bool IsAnonymous;
+
+        private List<string> bookList;
+
+        public string this[int index]
+        {
+            get
+            {
+                if (index >= this.BookCount || index >= this.BookLimit || index < 0)
+                    throw new IndexOutOfRangeException();
+                return this.bookList[index];
+            }
+            set
+            {
+                if (index >= this.BookCount || index >= this.BookLimit || index < 0)
+                    throw new IndexOutOfRangeException();
+                this.bookList[index] = value;
+            }
+        }
+
+        public LibraryUser()
+        {
+            this.FirstName = "No name";
+            this.LastName = "No last name";
+            this.Phone = "No phone";
+            this.Anonymous = true;
```

Comment:
variable Anonymous-  where does he write? and 65 line

Reply:
In task was said: implement two constructors with parameters and without. I think, only anonymous user can be without name, last name and phone (parameterless constructor). And what do you mean by he? Please, clear your question.

Reply:
you don't have such a variable Anonymous

Reply:
Oh, forgot to rename it((( fixed in commit ec77f715fd12673612e57b26bfd3abc546a93783